jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -285,6 +289,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
+    private List<String> getPreferredEncodingOrder(HttpServletRequest request)
+    {
+        Enumeration<String> e = request.getHeaders(HttpHeader.ACCEPT_ENCODING.asString());
+        if (e==null || !e.hasMoreElements())
+            return emptyList();
+
+        String key=e.nextElement();
+        while (e.hasMoreElements())
+            key+='\n'+e.nextElement();
```

Comment:
This is still a lot of work to do and garbage created to create a key to look into the cache.
You'd hope the optimizer would make this a StringBuilder and perhaps the most frequent case is only a single element...